scripts/histmakers/mz_wlike_with_mu_eta_pt.py

```diff
@@ -26,6 +26,11 @@
 
 era = args.era
 
+if args.IsoEfficiencysmoothing:
+    DirectIsoSFsmoothing = False
```

Comment:
Normally reserve capitalised variables for for class names. And why isn't this just directIsoSFsmoothing = args.IsoEfficiencysmoothing ?

Reply:
because it's the opposite of that, now we want to set sf smoothing as default, so we pass an argument in case we instead want to use eff smoothing, and the rest of the classes instead uses the flag which is true in case we want to use direct iso sf smoothing, so not to modify those I just did that

Reply:
Then it can be directIsoSFsmoothing = not args.IsoEfficiencysmoothing, if it is more convenient to call them differently.
You can also define the option as
parser.add_argument("--doThis", dest="dontDoThis", action="store_false", help="Do this (don't by default)")
So that the option triggers an action but the variable it touches would represent the opposite (and is true by default, meaning it doesn't do the action if not requested)

Reply:
Ok now I corrected everything so that it just uses isoEfficiencySmoothing in store_true, I changed muon_efficiency_smooth.py

Reply:
So here it would be --doEffiSmoothing and dest="doSFsmoothing"